rtl/eclrtl/CMakeLists.txt

```diff
@@ -68,7 +66,15 @@ include_directories (
          ${BOOST_REGEX_INCLUDE_DIR}
     )
 
-ADD_DEFINITIONS( -D_USRDLL -DECLRTL_EXPORTS )
+if (UNIX AND USE_BOOST_REGEX)
+    if (EXISTS "/etc/debian_version")
+#       message("-- Boost_VERSION: ${Boost_VERSION}")
+        if (${Boost_VERSION} LESS 104800)
+            message("---- using special flag for some sources due to older boost version")
+            set_source_files_properties(eclrtl.cpp PROPERTIES COMPILE_FLAGS -std=c++98)
```

Comment:
Just make this line unconditional and all the concerns about trying to make the dependencies conditional go away...